pouchdb.d.ts

```diff
@@ -1002,6 +1002,62 @@ declare module pouchdb {
                 }
             }
 
+            /** Contains the method and call/return types for query() */
+            module query {
+
+                interface ReduceKey {
+                    0: any,
+                    1: string
+                }
+
+                type MapFunction = (doc, emit) => void;
+                type ReduceFunction = <T>(keys: ReduceKey[], values: T[], rereduce: boolean) => T;
+
+                interface MapReduce<T> {
+                    map: MapFunction;
+                    reduce: ReduceFunction;
+                }
+
+                type QueryFunction<T> = MapFunction | MapReduce<T> | string;

```

Comment:
We need to split this into 2 categories, one using standalone map function (which receives a defined and typed `emit` function), and another one that uses functions that need to be aware of the implicitly available `emit` function. See coment above for details.